api/apps/api/src/modules/scenarios/project-checker/project-checker.service-real.ts

```diff
@@ -8,19 +8,46 @@ import { Project } from '@marxan-api/modules/projects/project.api.entity';
 import { InjectRepository } from '@nestjs/typeorm';
 import { PlanningAreasService } from '@marxan-api/modules/planning-areas/planning-areas.service';
 import { isDefined } from '@marxan/utils';
-
-export const doesntExist = Symbol(`doesn't exist`);
-export type DoesntExist = typeof doesntExist;
+import {
+  doesntExist,
+  DoesntExist,
+  hasPendingExport,
+  HasPendingExport,
+  ProjectChecker,
+} from '@marxan-api/modules/scenarios/project-checker/project-checker.service';
 
 @Injectable()
-export class ProjectChecker {
+export class ProjectCheckerReal implements ProjectChecker {
```

Comment:
surely minor - I'd rather keep class names descriptive, and the `Real` suffix here is a bit of a last resort... and I know things can get a bit ugly when an abstract class is such not so much because of actual abstract-ness but to provide a way to use the same interface for test fake classes, but what about just

```suggestion
export class ProjectChecker implements ProjectValidator {
```
or
```suggestion
export class ProjectChecker implements ProjectCheckerRules {
```

or something like this? It's all quite arbitrary/opinionated, but I'd save the most descriptive and plain name for the "real" class.

Reply:
This issue is always a pain 😂 
I've always preferred to have the most straighforward naming for the interface/abstract class, as it's the one that will be appearing more times in the codebase. `ProjectCheckerReal` can be improved but, as an example with repositories:
Abstract class: `ProjectRepository`
In memory implementation: `MemoryProjectRepository`
TypeORM implementation: `TypeORMProjectRepository` 
In this example (for me) seems the mos correct way of naming, as I see a little bit confusing to have `ProjectChecker` implementing `ProjectValidator`.
Returning to the original naming, I think the best way to handle this is having the abstract class as `ProjectChecker` and looking for a better name for the real implementation that represents that it is that, the actual implementation.

Reply:
agree that it's always a pain - though sometimes it can also be a (code) smell 😅 

thing is, in the kind of use we're making here and elsewhere, it's not like (in pseudocode) Dog implements abstract Animal.

I'd suggest `MarxanProjectChecker` for the "real" one maybe?

Reply:
Mmm ok, that'd be good 👌🏽